lib/decompress/zstd_decompress_block.c

```diff
@@ -1043,7 +1055,9 @@ ZSTD_decompressSequences_body( ZSTD_DCtx* dctx,
     /* last literal segment */
     {   size_t const lastLLSize = litEnd - litPtr;
         RETURN_ERROR_IF(lastLLSize > (size_t)(oend-op), dstSize_tooSmall, "");
-        memcpy(op, litPtr, lastLLSize);
+        if (op != NULL) {
+            memcpy(op, litPtr, lastLLSize);
+        }
         op += lastLLSize;
```

Comment:
Should this `op +=` operation belong to the `if (op...` branch too ?

Reply:
Fixed here and in legacy impls

Reply:
Well, strangely, I do see the _other_ fixes, 
but not the one on this line specifically ...

Reply:
Ah, I fixed the `decompressSequences_long()` one, but didn't fix this one.